### ISSUE TITLE
Learning rate is "constant" after the warmup step in cosine annealing

### ISSUE THREAD

**Issue by sravan500** [NVIDIA-NeMo/NeMo#14358]
We are working on automodel script for peft finetuning.
Observed that the learning rate dose not increase and constant after the "warmup_steps=50".
Please find my config:

optimizer = fdl.build(pytorch_adam_with_cosine_annealing(
    max_lr=1e-4,
    weight_decay=0.01,
    warmup_steps=50
)

<img width="1150" height="628" alt="Image" src="https://github.com/user-attachments/assets/90ba15ad-c644-49df-9a9c-350840faeda6" />

**Comment by akoumpa** [comment]
Thanks @sravan500 ,

I think this is a known issue unfortunately. If you are interested in the automodel path, please consider using our newer code available on https://github.com/NVIDIA-NeMo/Automodel. 

The new repo has correct support for cosine LR scheduler, so it's highly recommended it to try it.

Thank you.
